File: java/Runner.java
public class Runner {
    public static void main(String[] args) {

        Utility obj = new Child();

        String[] csvData = obj.readFileData();
        String[][] extractedData = new String[csvData.length][2];

        for (int i = 0; i < csvData.length; i++) {
            String otp = obj.extractOtp(csvData[i]);
            extractedData[i][0] = csvData[i];
            extractedData[i][1] = otp;
        }

        obj.writeFile(extractedData);
    }
}

File: javascript/Runner.js
const Child = require('./Child');

function main() {
    const obj = new Child();
    const csvData = obj.readFileData();
    
    // 2D array to store 3 cols indicating original_msg, extracted_otp and expected_otp
    const extractedData = new Array(csvData.length).fill(null).map(() => new Array(3));

    for (let i = 0; i < csvData.length; i++) {
        const message = csvData[i][0];
        const expected = csvData[i][1]; 
        const otp = obj.extractOtp(message);
        
        extractedData[i][0] = message;   
        extractedData[i][1] = otp;       
        extractedData[i][2] = expected;  
    }

    obj.writeFile(extractedData);
}

main();